riscv-c-api.md

```diff
@@ -621,3 +621,158 @@ statements, including both RISC-V specific and common operand modifiers.
 | ------------ | --------------------------------------------------------------------------------- | ----------- |
 | z            | Print `zero` (`x0`) register for immediate 0, typically used with constraints `J` |             |
 | i            | Print `i` if corresponding operand is immediate.                                  |             |
+
+## Function Multi-version
+
+Function multi-versioning (FMV) allows selecting the appropriate function based on the runtime environment. The binary may contain multiple versions of the function, with the compiler generating all supported versions and selecting the appropriate one during the runtime.
+
+This feature is activated by the `target_version/target_clones` function attributes.
+
+### Extension Bitmask
+
+When generating the resolver function for FMV, a method is required to retrieve environment information. This is achieved through two bitmask structures: `__riscv_feature_bits` for standard extensions and `__riscv_vendor_feature_bits` for vendor-specific extensions.
+
+The bitmask structures use the following definitions:
+
+```c
+#define MAXLENGTH TO_BE_DECIDED
+
+struct {
+    unsigned length;
+    unsigned long long features[MAXLENGTH];
+} __riscv_feature_bits;
+
+struct {
+    unsigned vendorID
+    unsigned length;
+    unsigned long long features[MAXLENGTH];
+} __riscv_vender_feature_bitmask;
+```
+
+- `length`: Represents the number of elements in the features array.
+- `features`: An `unsigned long long` array where each bit indicates the 1 for enable or 0 for disable of a corresponding extension.
+- `vendorID`: Indicates the current vendor core.
+
+To initiate these structures based on the system's extension status, the following function is provided:
+
+```
+void __init_riscv_features_bit();
+```
+
+The `__init_riscv_features_bit` function updates `length`, `vendorID` and the `features` in `__riscv_feature_bits` and `__riscv_vendor_feature_bits` according to the enabled extensions in the system.
+
+Each queryable extension must have an associated `groupid` and `bitmask` that indicates its position within the features array. 
+
+> For example, the zba extension is represented by `groupid`: 3 and `bitmask`: `1ULL << 5`. Users can check if the zba extension is enabled using: `__riscv_feature_bits.features[3] & (1ULL << 5)`.
```

Comment:
Isn't zba's groupid 0 and bitmask (1 << 52) in the table below?
If not, how do we decode the table below?